bitcoin/src/address/mod.rs

```diff
@@ -800,20 +755,20 @@ impl FromStr for Address<NetworkUnchecked> {
 
         let inner = match *prefix {
             PUBKEY_ADDRESS_PREFIX_MAIN => {
-                let hash = PubkeyHash::from_byte_array(data);
-                AddressInner::P2pkh { hash, prefix: LegacyP2pkhPrefix::Mainnet }
+                let hash = PubkeyHash::from_slice(&data).expect("length checked above");
```

Comment:
Why? It was previously statically correct.

Reply:
Yes, no idea what I was thinking when I did this. Fixed.